src/scripts/directives.js

```diff
@@ -293,6 +293,21 @@
         }
     ]);
 
+    module.filter('numberPadding', [
+        function () {
+            return _.memoize(function (input, numOfDigits) {
+                input = input.toString();
+                while (input.length < numOfDigits) {
+                    input = '0' + '' + input;
```

Comment:
You don't need `+ ''` here, since `'0'` will already coerce the expression to a string.